src/main/java/com/mongodb/hibernate/internal/translate/MongoTranslatorFactory.java

```diff
@@ -26,7 +26,14 @@
 import org.hibernate.sql.model.ast.TableMutation;
 import org.hibernate.sql.model.jdbc.JdbcMutationOperation;
 
+/**
+ * Tread-safe.
+ */
 public final class MongoTranslatorFactory implements SqlAstTranslatorFactory {
```

Comment:
Hibernate ORM does not document that implementations of `SqlAstTranslatorFactory` must be thread-safe, so we should document that for our `MongoTranslatorFactory`.

Reply:
For such bootstrapping stuff, usually no need to sepcify type safety for everything will be done sequentially with well-defined dependency or order. There is no need of concurrency for bootstrapping only happens once and its  perf could be compromised a little bit.

Reply:
sorry, I made a mistake here. Yeah, I think `SqlAstTranslatorFactory` should be thread-safe for it will be invoked concurrently.

You are right. This doc is good to have.

Reply:
Can this be resolved?